### PR TITLE
Created File Manager to localize FS changes.

### PR DESCRIPTION
**Pull request by MartinMueller2003**
Moved all FPP file handling to the new file manager
Moved file upload support to the file manager
Moved configuration file support from FileIO to File Manager
Moved setting values from Json to a template in ESPIxelstick.h

### CODE REVIEW

**Comment by forkineye** [comment]
Travis failed. Looks like it's just a capitalization error unless you're using a different version of the library that's not checked in? https://travis-ci.org/github/forkineye/ESPixelStick/builds/751210789#L2284

File: src/FileMgr.cpp [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+*  The Author makes no warranty of any kind, express or implied, with regard
+*  to this program or the documentation contained in this document.  The
+*  Author shall not be liable in any event for incidental or consequential
+*  damages in connection with, or arising out of, the furnishing, performance
+*  or use of these programs.
+*
+*/
+
+#include "ESPixelStick.h"
+#include <int64String.h>
```
Comment (forkineye) [inline src/FileMgr.cpp:21]: Should be <Int64String.h> ?

**Review by forkineye — COMMENTED** [review]
Is OutputMgr supposed to be using FileMgr now as well?  Notice FileIO references still in it.

**Comment by MartinMueller2003** [comment]
No. FileIO has been removed. I will check again to see if I missed anything.
Martin
    On Thursday, December 24, 2020, 09:11:48 AM EST, Shelby Merrick <notifications@github.com> wrote:  
 
 


@forkineye commented on this pull request.

Is OutputMgr supposed to be using FileMgr now as well? Notice FileIO references still in it.

In src/FileMgr.cpp:
> +* http://www.forkineye.com
+*
+*  This program is provided free for you to use in any way that you wish,
+*  subject to the laws and regulations where you are using it.  Due diligence
+*  is strongly suggested before using this code.  Please give credit where due.
+*
+*  The Author makes no warranty of any kind, express or implied, with regard
+*  to this program or the documentation contained in this document.  The
+*  Author shall not be liable in any event for incidental or consequential
+*  damages in connection with, or arising out of, the furnishing, performance
+*  or use of these programs.
+*
+*/
+
+#include "ESPixelStick.h"
+#include <int64String.h>

Should be <Int64String.h> ?

**Comment by MartinMueller2003** [comment]
Yup. Output did have one remaining instance of FileIO.
Fix will be in the next push
Martin

    On Thursday, December 24, 2020, 11:38:37 AM EST, Martin Mueller <martinmueller2003@yahoo.com> wrote:  
 
  No. FileIO has been removed. I will check again to see if I missed anything.
Martin
    On Thursday, December 24, 2020, 09:11:48 AM EST, Shelby Merrick <notifications@github.com> wrote:  
 
 


@forkineye commented on this pull request.

Is OutputMgr supposed to be using FileMgr now as well? Notice FileIO references still in it.

In src/FileMgr.cpp:
> +* http://www.forkineye.com
+*
+*  This program is provided free for you to use in any way that you wish,
+*  subject to the laws and regulations where you are using it.  Due diligence
+*  is strongly suggested before using this code.  Please give credit where due.
+*
+*  The Author makes no warranty of any kind, express or implied, with regard
+*  to this program or the documentation contained in this document.  The
+*  Author shall not be liable in any event for incidental or consequential
+*  damages in connection with, or arising out of, the furnishing, performance
+*  or use of these programs.
+*
+*/
+
+#include "ESPixelStick.h"
+#include <int64String.h>

Should be <Int64String.h> ?

**Pull request merged**